examples/tensorflow_mnist.py

```diff
@@ -62,18 +67,46 @@ def conv_model(feature, target, mode):
     return tf.argmax(logits, 1), loss
 
 
+def train_input_generator(x_train, y_train, batch_size=64):
+    assert len(x_train) == len(y_train)
+    while True:
+        p = np.random.permutation(len(x_train))
+        x_train, y_train = x_train[p], y_train[p]
+        index = 0
+        while index <= len(x_train) - batch_size:
+            yield x_train[index:index + batch_size], \
+                  y_train[index:index + batch_size],
+            index += batch_size
+
+
 def main(_):
     # Horovod: initialize Horovod.
     hvd.init()
 
     # Download and load MNIST dataset.
-    mnist = learn.datasets.mnist.read_data_sets('MNIST-data-%d' % hvd.rank())
+    try:
+        (x_train, y_train), (x_test, y_test) = \
+            keras.datasets.mnist.load_data('MNIST-data-%d' % hvd.rank())
+    except OSError as ex:
+        # When running tests, if dataset is previously downloaded, it may cause
+        # the tests to fail. In this case, we need to remove the dataset cache
+        # folder first and download the dataset again.
+        cache_dir = os.path.join(os.path.expanduser('~'), '.keras')
+        datadir_base = os.path.expanduser(cache_dir)
+        datadir = os.path.join(datadir_base, "datasets")
+        shutil.rmtree(datadir)
+        (x_train, y_train), (
+            x_test, y_test) = keras.datasets.mnist.load_data(
+            'MNIST-data-%d' % hvd.rank())
+
+    x_train = np.reshape(x_train, (-1, 784)) / 255
```

Comment:
Is this safe in Python2, where integer division is the default?  Maybe we can divide by 255.0 to be safe?

Reply:
fixed.